lib/explorer/series.ex

```diff
@@ -2244,6 +2244,51 @@ defmodule Explorer.Series do
   @spec pow(left :: Series.t() | number(), right :: Series.t() | number()) :: Series.t()
   def pow(left, right), do: basic_numeric_operation(:pow, left, right)
 
+  @doc """
+  Calculate the exponent of a logarithm.
+
+  The resultant series is going to be of dtype :integer
+  if the series and the base are both integers. Otherwise
+  the it is going to be a float series.
+
+  ## Supported dtypes
+
+    * `:integer`
+    * `:float`
+
+  ## Examples
+
+      iex> s = Explorer.Series.from_list([8, 16, 32])
+      iex> Explorer.Series.log(s, 2)
+      #Explorer.Series<
+        Polars[3]
+        integer [3, 4, 5]
+      >
+
+      iex> s = Explorer.Series.from_list([8, 16, 32])
+      iex> Explorer.Series.log(s, 2.0)
+      #Explorer.Series<
+        Polars[3]
+        float [3.0, 4.0, 5.0]
+      >
+
+  """
+  @doc type: :element_wise
+  @spec log(argument :: Series.t(), base :: number()) :: Series.t()
+  def log(argument, base \\ 2) when is_number(base) do
```

Comment:
For compatibility with Nx, the default should be a natural logarithm (where the base is `e`). My suggestion is to have `log/1` and `log/2` as distinct functions.